### PR TITLE
Add ability to let addons generate taskfile sections

### PR DESCRIPTION
**Pull request by rick-nu**
# What

Add ability to let addons generate Taskfile sections

## Why

We want to add many generated sections to the taskfile. Each addon should be able to add related commands/functions/lines to the desired taskfile sections. This PR adds the ability to fill those sections per addon.

### CODE REVIEW

File: src/components/Generator/Generator.tsx [hunk 1]
```diff
@@ -7,12 +7,13 @@ import styles from './generator.module.css';
...
 import GeneratedTaskfile from './GeneredTaskfile';
-import Form from "@/components/Form";
-import {Font} from "./GeneredTaskfile/buildHeader";
+import Form from '@/components/Form';
+import { Font } from './GeneredTaskfile/buildHeader';
 
 export type GeneratorSettings = {
 	project: string;
 	font: Font;
+	runtime: 'local' | 'docker-compose' | 'docker-compose-dev';
```
Comment (rick-nu) [inline src/components/Generator/Generator.tsx]:
```suggestion
	runtime: 'local' | 'docker-compose';
```

Make a separate checkbox for including the development proxy when `docker-compose` is chosen

File: src/components/Generator/GeneredTaskfile/addons/runtime.ts
Comment (rick-nu) [inline src/components/Generator/GeneredTaskfile/addons/runtime.ts:1]: Rename to docker to be more clear

File: src/components/Generator/GeneredTaskfile/taskfile.ts [hunk 2]
```diff
@@ -1,7 +1,34 @@
...
+		initFunctionPrepend: [],
+		projectSection: [],
+		startFunction: [],
+		updateFunction: [],
+		customSections: [],
+		utilitySection: [],
+		globals: [],
+	}
+
+	buildAddonRuntime(settings, addons);
```
Comment (rick-nu) [inline src/components/Generator/GeneredTaskfile/taskfile.ts:29]:
```suggestion
	buildAddonDocker(settings, addons);
```

**Comment by rick-nu** [comment]
@johanvo this is now ready for review!

**Comment by rick-nu** [comment]
https://taskfilegenerator-qbr13gai9-taskfile-generators-projects.vercel.app/

**Comment by rick-nu** [comment]
https://taskfilegenerator-6cuv9dgpg-taskfile-generators-projects.vercel.app/

File: src/helpers/loadTemplate.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * @param file should be an imported bash (.sh) file
+ * @param variables an object with all variables that you want to replace
+ * @example
+ * import myAddonSection from './my-addon-section.sh';
+ *
+ * // Replaces `[[foo]]` with `bar` in the template
+ * loadTemplate(myAddonSection, {foo: 'bar'});
+ */
+const loadTemplate = (file: string, variables?: TemplateVariables): string => {
```
Comment (johanvo) [inline src/helpers/loadTemplate.ts:15]: Super nice man, precies zoals ik hoopte :clap: :-D

**Review by johanvo — APPROVED** [review]
Meest kritieke dingen doorgekeken en die vond ik prachtig! De rest kijk ik later wel, wil het mergen hiervan niet langer ophouden dus lets go! :clap:

**Pull request merged**